### DISCUSSION TITLE
If a new variable is added after the state is updated and not defined in the state at first, only $state can obtain it

### DISCUSSION THREAD

**Discussion by vela666** [vuejs/pinia#1783] [Help and Questions] [answered]
Can I only use $state or define variables in advance?

![image](https://user-images.githubusercontent.com/69613314/200520546-79232313-94ad-4777-bf8d-493d60cdacd0.png)
![image](https://user-images.githubusercontent.com/69613314/200520758-39fc398f-661c-46f2-9ec0-c15c99c349cf.png)
![image](https://user-images.githubusercontent.com/69613314/200520957-b1edab2e-b07b-475e-ab01-6ccec40a9af7.png)

**Comment by posva** [comment, marked as answer]
Yes, this is normal: you have to declare your initial state.

If for some reason, that's not possible to you, you have to manually declare both properties to access it diretly on the store. You can find an example at https://pinia.vuejs.org/core-concepts/plugins.html#adding-new-state.

Note you can also keep your way and only access the property through `store.$state` if that is fine for you.

**Reply by vela666** [reply]
okay, thank you

**Reply by vela666** [reply]
> 好的谢谢你

In this way https://pinia.vuejs.org/core-concepts/plugins.html#adding-new-state  
Why are all the data added to the main? How can I specify the location I want

![image](https://user-images.githubusercontent.com/69613314/201640185-a886442c-e82d-4ab9-b2d3-4020afddb5f2.png)

**Reply by posva** [reply]
```ts
pinia.state.value.user === userStore.$state
```

it depends on what store's `$state` you set. Each one is associated with one key inside `pinia.state.value`
